cppForSwig/BlockDataMap.cpp

```diff
@@ -89,7 +123,7 @@ void BlockData::deserialize(const uint8_t* data, size_t size,
    vector<BinaryData> allhashes;
    for (auto& txn : txns_)
    {
-      BinaryDataRef txdata(txn->data_, txn->size_);
+      BinaryDataRef txdata(txn->noWitData_);
```

Comment:
notice how the merkle computation depends on the checkMerkle flag. The merkle is only checked at build for block consistency, it is entirely skipped during scans (we trust the integrity of the db)